### DISCUSSION TITLE
No model for attachments in graphql_data_is_private

### DISCUSSION THREAD

**Discussion by maurocolella** [wp-graphql/wp-graphql#2481] [General] [answered]
Hi,

In this hook, while the item data includes a `post_type` that lets me distinguish attachment from posts (or pages), attachment always report as: `PostObject`

Should they have their own model?

**Comment by justlevine** [comment, marked as answer]
Hey @maurocolella ,

As you noted, `$data->post_type` should show `attachment`. 

Typically a `Model` describes how the data is _shaped_ and not the content that is stored inside of it, and [attachments are a `WP_Post_Type`](https://developer.wordpress.org/themes/basics/post-types/#attachment) hence the model name.

To distinguish attachments from other post types in `graphql_data_is_private`, you can do the following:

```php
add_filter(
  'graphql_data_is_private',
  function( $is_private, $model_name, $data, $visibility, $owner, $current_user ) {
    if( 'PostObject' === $model_name && 'attachment' === $data->post_type ) {
      // return some check based on the $owner and $current_user
    }
    // For anything else..
    return $is_private;
  },
  10,
  6
);
```

Let us know if that solves things for you, or if you have any further questions :-)
